src/test/java/com/google/cloud/spanner/pgadapter/PgAdapterTestEnv.java

```diff
@@ -105,30 +107,36 @@
   // Spanner options for creating a client.
   private SpannerOptions options;
 
+  // Shared Spanner instance that is automatically created and closed.
+  private Spanner spanner;
+
   // Spanner URL.
   private URL spannerURL;
 
   // Log stream for the test process.
   private static final Logger logger = Logger.getLogger(PgAdapterTestEnv.class.getName());
 
-  // Utility class for setting up test connection.
-  private RemoteSpannerHelper spannerHelper;
-
   private final List<Database> databases = new ArrayList<>();
 
   public void setUp() throws Exception {
     spannerURL = new URL(getHostUrl());
     options = createSpannerOptions();
   }
 
-  public SpannerOptions spannerOptions() {
-    return options;
+  public Spanner getSpanner() {
+    if (spanner == null) {
```

Comment:
Create a single shared `Spanner` instance instead of returning `SpannerOptions`. The `Spanner` instance is closed when the test environment is cleaned up.